e2e_tests.py

```diff
@@ -93,95 +109,120 @@ def invalidMap(tag, method, path, bodies):
+        ['logout', 'post', '/auth/logout', {}, {}, 200],
+        ['check that session is no longer valid', 'get', '/profile', {}, '', 403],
+        ['login again', 'post', '/auth/login', {}, {'username': username, 'password': 'foobar2'}, 200, successfulLogin],
+        invalidMap ('change password', 'post', '/auth/change_password', ['', [], {}, {'foo': 'bar'}, {'old_password': 1}, {'old_password': 'foobar'}, {'old_password': 'foobar', 'new_password': 1}]),
+        ['get profile before profile update', 'get', '/profile', {}, {}, 200, getProfile1],
+        invalidMap ('update profile', 'post', '/profile', ['', [], {'email': 'foobar'}, {'birth_year': 'a'}, {'birth_year': 20}, {'country': 'Netherlands'}, {'gender': 0}, {'gender': 'a'}]),
+        ['change profile with same email', 'post', '/profile', {}, {'email': username + '@e2e-testing.com', 'country': 'US'}, 200],
+        ['change profile with different email', 'post', '/profile', {}, {'email': username + '@e2e-testing2.com', 'country': 'NL'}, 200, emailChange],
+        ['get profile after profile update', 'get', '/profile', {}, {}, 200, getProfile2],
+        ['verify email after email change', 'get', lambda state: '/auth/verify?' + urllib.parse.urlencode ({'username': username, 'token': state ['token2']}), {}, '', 302],
+        ['get profile after email verification', 'get', '/profile', {}, {}, 200, getProfile3],
+        invalidMap ('recover password', 'post', '/auth/recover', ['', [], {}, {'username': 1}]),
+        ['recover password, invalid user', 'post', '/auth/recover', {}, {'username': 'nosuch'}, 403],
+        ['recover password', 'post', '/auth/recover', {}, {'username': username}, 200, recoverPassword],
+        invalidMap ('reset password', 'post', '/auth/reset', ['', [], {}, {'username': 1}, {'username': 'foobar', 'token': 1}, {'username': 'foobar', 'token': 'some'}, {'username': 'foobar', 'token': 'some', 'password': 1}, {'username': 'foobar', 'token': 'some', 'password': 'short'}]),
+        ['reset password, invalid username', 'post', '/auth/reset', {}, lambda state: {'username': 'foobar', 'token': state ['token'], 'password': 'foobar'}, 403],
+        ['reset password, invalid token', 'post', '/auth/reset', {}, lambda state: {'username': username, 'token': 'foobar', 'password': 'foobar'}, 403],
+        ['reset password, invalid password', 'post', '/auth/reset', {}, lambda state: {'username': username, 'token': state ['token'], 'password': 'short'}, 400],
+        ['reset password', 'post', '/auth/reset', {}, lambda state: {'username': username, 'token': state ['token'], 'password': 'foobar3'}, 200],
+        ['login again after reset', 'post', '/auth/login', {}, {'username': username, 'password': 'foobar3'}, 200],
+        ['destroy account', 'post', '/auth/destroy', {}, {}, 200]
+    ]
+
+if len (sys.argv) == 1:
```

Comment:
Switching to [`argparse`](https://docs.python.org/3/library/argparse.html) might be more readable, maintainable *and* usable rather than having unexplained command line behaviors.

Reply:
Again, I'd rather not add dependencies for very simple functions. But I could document within the tester what parameters it expects so its usage is crystal clear.

Reply:
`argparse` is built into the standard library so it's not much of an additional dependency.

Reply:
OK, I will use argparse.